### PR TITLE
feat: integrate IDR with mock app

### PR DESCRIPTION
**Pull request by namhoang1604**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR integrate the IDR with the mock app

## Related Tickets & Documents

https://github.com/gs-gs/fa-ag-trace/issues/561

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: packages/services/src/utils/helpers.ts [hunk 1]
```diff
@@ -257,3 +257,8 @@ export const constructObject = (
 
   return data;
 };
+
+export const extractDomain = (url: string): string => {
```
Comment (ldhyen99) [inline packages/services/src/utils/helpers.ts:261]: From my understanding, if we do not have try/catch when calling `new URL`, the default error is `Throws TypeError: Invalid URL`. Should we custom this error?

Comment (namhoang1604) [reply packages/services/src/utils/helpers.ts:261]: No, we shouldn't

File: packages/services/src/utils/helpers.ts [hunk 2]
```diff
@@ -301,6 +301,11 @@ export const constructObject = (
   return data;
 };
 
+export const extractDomain = (url: string): string => {
```
Comment (ldhyen99) [inline packages/services/src/utils/helpers.ts:304]: Can you provide the unit test of this function?

Comment (namhoang1604) [reply packages/services/src/utils/helpers.ts:304]: Good catch!

**Review by ldhyen99 — DISMISSED** [review]

File: packages/services/src/linkResolver.service.ts [hunk 3]
```diff
@@ -96,46 +101,51 @@ export interface GS1LinkResponse extends ILinkResponse {
...
 };
 
 export const constructLinkResolver = (
+  namespace: string,
   linkResolver: ILinkResolver,
   linkResponses: ILinkResponse[],
   qualifierPath: string,
 ) => {
-  const gs1LinkResolver: GS1LinkResolver = {
+  const LinkResolver: LinkResolver = {
```
Comment (huynguyen-hl) [inline packages/services/src/linkResolver.service.ts:137]: I think the `LinkResolver` variable should follow the camelCase naming convention.

File: packages/components/src/components/BarcodeGenerator/BarcodeGenerator.tsx [hunk 4]
```diff
@@ -24,7 +24,38 @@ export const BarcodeGenerator = (props: IBarcodeProps) => {
...
+        const parsedValues = values.map((item: any) => {
+          const convertToGS1String = (obj: any) => {
+            if (typeof obj !== 'object' || obj === null) return '';
+
+            let result = '';
+
+            // Handle '01' (01) first if it exists
+            if ('01' in obj) {
+              let value = obj['01'];
+              if (!/^\d{8,14}$/.test(value)) throw new Error('GTIN must be 8-14 digits');
```
Comment (huynguyen-hl) [inline packages/components/src/components/BarcodeGenerator/BarcodeGenerator.tsx:36]: The regex hardcoded in the `if` statement to check the `01` AI value here might be incorrect. I checked the GS1 codebase and noticed they use this regex `(\d{12,14}|\d{8})` to validate the `01` AI value. You may  double-check it to ensure it's correct.

Comment (namhoang1604) [reply packages/components/src/components/BarcodeGenerator/BarcodeGenerator.tsx]: I checked and tested again, and both cases worked well but I changed to follow the GS1 regex

**Review by huynguyen-hl — APPROVED** [review]
LGTM

**Pull request merged**